django_advanced_queries/covid_19/tests.py

```diff
@@ -356,48 +358,60 @@ def setUp(self):
         with self.assertNumQueries(1):
-            num_of_hospitalized_because_of_botism_or_corona = Patient.objects.filter_by_examinations_results_options(
-                results=('Botism', 'Corona')
-            ).count()
+            num_of_hospitalized_because_of_botism_or_corona = Patient.objects.filter(
+                medical_examination_results__result__in=[
+                    MedicalExaminationResult.RESULT_BOT,
+                    MedicalExaminationResult.RESULT_CORONA]
+            ).distinct().count()
+
             self.assertEqual(num_of_hospitalized_because_of_botism_or_corona, 7)
 
     def test_hospital_worker_alon_age_using_single_query(self):
         with self.assertNumQueries(1):
-            alon = HospitalWorker.objects.get(person__name='Alon')
+            alon = HospitalWorker.objects.select_related('person').get(
+                person__name='Alon')
             self.assertEqual(alon.person.age, 65)
 
     def test_count_all_hospital_departments_using_two_queries(self):
+        # could do that with 1 query (using annotate)
         with self.assertNumQueries(2):
-            hospitals = Hospital.objects.all()
+            hospitals = Hospital.objects.all().prefetch_related('departments')
             for hospital in hospitals:
                 self.assertEqual(hospital.departments.count(), 1)
 
     def test_highest_num_of_patient_medical_examinations(self):
         with self.assertNumQueries(1):
-            highest_num_of_patient_m_e = Patient.objects.get_highest_num_of_patient_medical_examinations()
+            highest_num_of_patient_m_e = Patient.objects.annotate(results=
+            Count(
+                'medical_examination_results')).aggregate(
+                max_results=
+                Max('results'))['max_results']
             self.assertEqual(highest_num_of_patient_m_e, 4)
 
     def test_average_age_of_patients_in_every_department(self):
         with self.assertNumQueries(1):
-            departments_with_avg_age_of_patients = Department.objects.annotate_avg_age_of_patients()
+            departments_with_avg_age_of_patients = Department.objects.annotate(
+                avg_age_of_patients=Avg('patients_details__person__age'))
 
             actual_result = [department.avg_age_of_patients
-                             for department in departments_with_avg_age_of_patients.order_by()]
+                             for department in
+                             departments_with_avg_age_of_patients.order_by()]
             self.assertEqual(actual_result, [36, 48.75])
 
     def test_doctor_performed_the_most_medical_examinations(self):
         with self.assertNumQueries(1):
-            doctor_performed_the_most_m_e = HospitalWorker.objects.get_worker_performed_most_medical_examinations(
-                filter_kwargs={'position': 'Doctor'},
-                exclude_kwargs={}
-            )
+            doctor_performed_the_most_m_e = HospitalWorker.objects.filter(
```

Comment:
keep the same arguments as used in the original test

Reply:
d